edx_event_bus_kafka/internal/producer.py

```diff
@@ -157,6 +167,39 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+class ProducingContext:
+    """
+    Wrapper class to allow us to link a call to produce() with the on_event_deliver callback
+    """
+    def __init__(self, context_data):
+        self.context_data = context_data
+
+    def __str__(self):
+        return " ".join([f"{key}={value}" for key, value in self.context_data.items()])
+
+    def set_context_data(self, key, value):
+        self.context_data[key] = value
+
+    def on_event_deliver(self, err, evt):
+        """
+        Simple callback method for debugging event production
+
+        If there is any error, log all the known information about the calling context so the event can be recreated
+        and/or replayed later
+
+        Arguments:
+            err: Error if event production failed
+            evt: Event that was delivered (or failed to be delivered)
+        """
+        if err is not None:
+            logger.warning(
```

Comment:
1. I think this should be an error. Why just a warning?
2. I'd like to add a call to `record_exception()`. This could be a "TODO" comment and new task. That call needs to be in the context of exception handling, so we might need to both raise a chained exception and catch it. Maybe a helper method would keep this looking clean?

Reply:
[fixed]